lib/author.rb

```diff
@@ -0,0 +1,25 @@
+class Author
+
+  attr_reader :first_name, :last_name, :name, :books
+
+  def initialize(author)
+    @first_name = author[:first_name]
+    @last_name = author[:last_name]
+    @name = "#{@first_name} #{@last_name}"
+    @books = []
+  end
+
+  def write(title, publication_date)
+    new_book = Book.new({
+      :author_first_name => @first_name,
+      :author_last_name => @last_name,
+      :title => title,
+      :publication_date => publication_date
+    })
+    # had to add the shovel here to get it working, want to discuss why later though
+    @books << new_book
+    new_book
```

Comment:
The reason that you needed to use the shovel operator here AND return the new book is because this method is designed to: 1. Create a new book, 2. Add that book to the author's books, and 3. Return the new book.

You have to do each of these steps in order to match the interaction pattern :)